File: collab-board-backend/src/main/java/com/example/collabboard/service/card/CardService.java
package com.example.collabboard.service.card;

import com.example.collabboard.dto.card.CardRequest;
import com.example.collabboard.dto.card.CardResponse;

import java.util.List;

public interface CardService {

    CardResponse createCard(String boardId, CardRequest request);
    List<CardResponse> getCardsByBoard(String boardId);
    List<CardResponse> getCardsByList(String listId);

    CardResponse updateCard(String cardId, CardRequest request);

    CardResponse moveCard(String cardId, String targetListId, Integer newPosition);

    void deleteCard(String cardId);
}

File: collab-frontend/src/app/core/api/card.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Card } from '../models/card.model';
import { Observable } from 'rxjs';

@Injectable({ providedIn: 'root' })
export class CardService {

  private readonly http = inject(HttpClient);

  getCardsForBoard(boardId: string): Observable<Card[]> {
    return this.http.get<Card[]>(`boards/${boardId}/cards`);
  }

  createCard(payload: {
    boardId: string;
    listId: string;
    title: string;
    description?: string;
    position: number;
  }): Observable<Card> {
    return this.http.post<Card>('cards', payload);
  }

  updateCard(cardId: string, payload: Partial<Card>): Observable<Card> {
    return this.http.put<Card>(`cards/${cardId}`, payload);
  }

  moveCard(cardId: string, targetListId: string, newPosition: number): Observable<Card> {
    return this.http.put<Card>(`cards/${cardId}/move`, null, {
      params: {
        targetListId,
        newPosition
      }
    });
  }

  deleteCard(cardId: string): Observable<void> {
    return this.http.delete<void>(`cards/${cardId}`);
  }
}
